Add IOU routes to Universal Links (AASA)

kidroca opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
Updated the AASA (apple-app-site-association) file with configuration for the IOU routes

### Fixed Issues
<!-- Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing -->
Fixes #2401

### Tests
1. Have the E.cash app installed
2. Open an IOU link like https://staging.expensify.cash/iou/split/reportID
3. The link should be handled in-app - it should launch the E.cash app or prompt you to use E.cash to open the link

### QA Steps
Same as above

### Tested On
I've tested primarily on iOS. 
Since I can't test this change locally, I've just verified the other platforms are still working

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
<!-- ![image](https://user-images.githubusercontent.com/12156624/118790375-5ed08700-b89e-11eb-8c22-445009b0e9da.png)  -->


#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->
<!-- ![image](https://user-images.githubusercontent.com/12156624/118790438-6f80fd00-b89e-11eb-8bdf-6587b82cb43f.png) -->

#### Desktop

<!-- ![image](https://user-images.githubusercontent.com/12156624/118789874-e36ed580-b89d-11eb-8f74-9888962992f3.png) -->

#### iOS

_The changes that I used to test this locally were reverted._

https://user-images.githubusercontent.com/12156624/118555332-3d22b300-b76b-11eb-8705-eb25417fda76.mp4

#### Android
<!-- Insert screenshots of your changes on the Android platform-->
<!-- ![image](https://user-images.githubusercontent.com/12156624/118791007-f6ce7080-b89e-11eb-866f-20f81f97dba1.png) -->

kidroca committed:
Update: handle IOU universal links (apple-app-site-association)

kidroca commented:
I couldn't find a way to test this locally from the dev environment

I created a local server where I hosted the AASA file and an html file with a link like https://staging.expensify.cash/iou/split but since my local server is not registered in associated domains here it does not work
![image](https://user-images.githubusercontent.com/12156624/118556093-203aaf80-b76c-11eb-89c0-f4a4f56df341.png)

I can't add my test server to **Associated Domains** to see if things work in dev, because in order to build I'd have to update the signing configuration, with my development account and then use a different bundle ID and then add this ID in AASA ...
Also you can't just add `applinks:localhost` in there, it has to be a domain. Most examples use `ngrok` 

---

I saw Storybook was recently added to the app, and I've allowed myself the liberty to place a "story" that allows you to manually test deep links (and Universal Links) 
I've added all the static links in one section
While the dynamic links - that require routing parameters can be added as separate items
You can see how this works from this video:

https://user-images.githubusercontent.com/12156624/118555332-3d22b300-b76b-11eb-8705-eb25417fda76.mp4

Of course if this doesn't look useful to you it's just my last commit and I can easily drop it

---

Whatever you decide, updating the staging environment with the new AASA file should resolve the issue

One thing I noticed that the provided link in the ticket is missing a reportID parameter - https://staging.expensify.cash/iou/split 
maybe that's just the text in the email, but the link needs to look like this in order to work in-app: https://staging.expensify.cash/iou/split/72207410

Here's a sample video with the link including and not including a report ID

https://user-images.githubusercontent.com/12156624/118557686-347fac00-b76e-11eb-8fe1-21450f289a7a.mov

Gonals commented:
Hey @marcaaron, requesting your review here since I'm not very familiar with IOUs in general

Gonals commented:
Hi @kidroca!
Thanks for taking the time to solve this issue! While the story for testing purposes looks interesting, it is certainly outside the scope of the fix, can you remove it from the PR?
That said, if you want, feel free to create a new issue with the idea of adding this story to our code. We'll evaluate it and, if we are interested, you'll be able to re-submit your code in a different PR.

Thanks!

kidroca commented:
Thanks, I've extracted it to a separate branch here: https://github.com/kidroca/Expensify.cash/commit/17a3800be36f2830920d409d5133550d43fcaee6

Will post a ticket about it when I get some free time

kidroca commented:
Updated 
Reverted the change related to storybook and the PR now contains only the AASA file change
Also I've tested that all platforms are still working as before

marcaaron reviewed (approved):
LGTM

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.
